lib/node_modules/@stdlib/_tools/scaffold/math-special-unary/data/docs/types/index__d__ts.txt

```diff
@@ -0,0 +1,146 @@
+/*
+* @license Apache-2.0
+*
+* Copyright (c) {{YEAR}} {{COPYRIGHT}}.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/// <reference types="@stdlib/types"/>
+
+import { typedndarray, {{NDARRAY_IMPORT_TYPES}}, genericndarray, RealAndGenericDataType as DataType, Order } from '@stdlib/types/ndarray';
+
+/**
+* Input array.
+*/
+type InputArray = {{INPUT_NDARRAY_TYPE}} | genericndarray<number>;
+
+/**
+* Output array.
+*/
+type OutputArray = {{OUTPUT_NDARRAY_TYPE}} | genericndarray<number>;
+
+/**
+* Interface describing options.
+*/
+interface Options {
+	/**
+	* Output array order.
+	*
+	* ## Notes
+	*
+	* -   By default, the order of the output array is the same as the input array.
+	*/
+	order?: Order;
+
+	/**
+	* Output array data type.
+	*/
+	dtype?: DataType;
+}
+
+/**
+* Interface describing a unary element-wise function.
+*/
+interface UnaryFunction {
+	/**
+	* {{MAIN_DESC}}
+	*
+	* @param x - input ndarray
+	* @param options - options
+	* @returns output ndarray
+	*
+	* @example
+	* var ndarray2array = require( '@stdlib/ndarray/to-array' );
+	* var array = require( '@stdlib/ndarray/array' );
+	*
+	* var x = array( {{JS_EXAMPLE_VALUES_2X2}} );
+	* // returns <ndarray>
+	*
+	* var y = {{ALIAS}}( x );
+	* // returns <ndarray>
+	*
+	* var arr = ndarray2array( y );
+	* // returns {{JS_EXAMPLE_RESULTS_2X2}}
+	*/
+	( x: InputArray, options?: Options ): typedndarray<number>; // FIXME: we lose type specificity here, as the output ndarray data type is determined according to the output data type policy in conjunction with the `dtype` option
```

Comment:
This isn't correct. E.g., what happens if a unary function returns a complex number? Meaning, `number` is not always correct. I don't have a great solution for this. We may need to do conditional overloads. Meaning, for those apis accepting and returning complex numbers, we may need an overload for when `x` is a complex array, as then the output should also be a complex array. In short, the structure of this file depends on the output dtype policy.